app/src/main/java/org/thoughtcrime/securesms/util/RoundedBackgroundSpan.kt

```diff
@@ -0,0 +1,59 @@
+package org.thoughtcrime.securesms.util
+
+import android.content.res.Resources
+import android.graphics.Canvas
+import android.graphics.Paint
+import android.graphics.RectF
+import android.text.style.ReplacementSpan
+
+/**
+ * A Span that draws text with a rounded background.
+ *
+ * @param textColor - The color of the text.
+ * @param backgroundColor - The color of the background.
+ * @param cornerRadius - The corner radius of the background in pixels. Defaults to 8dp.
+ * @param paddingHorizontal - The horizontal padding of the text in pixels. Defaults to 3dp.
+ * @param paddingVertical - The vertical padding of the text in pixels. Defaults to 3dp.
+ */
+
+
+class RoundedBackgroundSpan(
+    private val textColor: Int,
+    private val backgroundColor: Int,
+    private val cornerRadius: Float = toPx(8, Resources.getSystem()).toFloat(), // setting some Session defaults
+    private val paddingHorizontal: Float = toPx(3, Resources.getSystem()).toFloat(),
+    private val paddingVertical: Float = toPx(3, Resources.getSystem()).toFloat()
+) : ReplacementSpan() {
+
+    override fun draw(
+        canvas: Canvas, text: CharSequence, start: Int, end: Int,
+        x: Float, top: Int, y: Int, bottom: Int, paint: Paint
+    ) {
+        // the top needs to take into account the font and the required vertical padding
+        val newTop = y + paint.fontMetrics.ascent - paddingVertical
+        val newBottom = y + paint.fontMetrics.descent + paddingVertical
+        val rect = RectF(
+            x,
+            newTop,
+            x + measureText(paint, text, start, end) + 2 * paddingHorizontal,
```

Comment:
Would it be expensive to do measuring while drawing? I thought the measure should already be done at this point and the boundary passed in the `draw` method should be sufficient?

Reply:
The `draw` unfortunately does not provide any information on the width beyond giving you the charsequence, which is why it needs to be calculated there.

Reply:
My bad I didn't see that the `draw` doesn't give you this info.